### PR TITLE
Adding fxdependent PS package tests

### LINKED ISSUE #7750: Add tests for framework dependent package
Add tests to validate package generated by #7729. The package must be built and then installed on a machine having the dotnet runtime installed. Then `dotnet ` should be called on `pwsh.dll` that is built. 

Steps to reproduce
------------------

```powershell
dotnet pwsh.dll
```

Expected behavior
-----------------

```none
PowerShell console starts and can execute cmdlets.
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0-rc.1
PSEdition                      Core
GitCommitId                    6.1.0-rc.1
OS                             Microsoft Windows 10.0.18230
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by anmenaga**
## PR Summary

Fix #7750 

fxdependent package = .NET framework dependent PowerShell package.
It is is a small-footprint (~22 MB) PowerShell package that can be run on a machine where dotnet runtime is installed separately. Requirement for having such package is driven by .NET team.

This PR adds release-time tests for fxdependent package for all currently supported Linux configurations.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [X] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: docker/tests/Templates/fxdependent-centos7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+RUN localedef --charmap=UTF-8 --inputfile=en_US $LANG
+
+# Install PowerShell package
+RUN curl -L -o $PACKAGENAME $PACKAGELOCATION/$PACKAGENAME
+RUN mkdir -p /opt/microsoft/powershell
+RUN tar zxf $PACKAGENAME -C /opt/microsoft/powershell
+
+# Download and run tests
+RUN $TESTDOWNLOADCOMMAND
+RUN dotnet /opt/microsoft/powershell/pwsh.dll -c "Import-Module /PowerShell/build.psm1;\$dir='/usr/local/share/powershell/Modules';\$null=New-Item -Type Directory -Path \$dir -ErrorAction SilentlyContinue;Restore-PSPester -Destination \$dir;exit (Invoke-Pester $TESTLIST -PassThru).FailedCount"
```
Comment (TravisEz13) [inline docker/tests/Templates/fxdependent-centos7/Dockerfile]: missing newline at end of the file

File: docker/tests/Templates/fxdependent-centos7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM centos:7
```
Comment (TravisEz13) [inline docker/tests/Templates/fxdependent-centos7/Dockerfile]: We should base this off the 2.1-runtime and 2.1-sdk image flavors from https://hub.docker.com/r/microsoft/dotnet/tags/

File: docker/tests/Templates/fxdependent-debian.8/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Install dependencies
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+        apt-utils \
+        ca-certificates \
+        curl \
+        apt-transport-https \
+        locales \
+        git \
+        wget
```
Comment (TravisEz13) [inline docker/tests/Templates/fxdependent-debian.8/Dockerfile]: don't add `wget`, use the docker `ADD` commande

File: docker/tests/Templates/fxdependent-centos7/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+ARG PACKAGENAME
+ARG PACKAGELOCATION
+ARG PREVIEWSUFFIX=
+ARG TESTLIST=/PowerShell/test/powershell/Modules/PackageManagement/PackageManagement.Tests.ps1,/PowerShell/test/powershell/engine/Module
+ARG TESTDOWNLOADCOMMAND="git clone --recursive https://github.com/PowerShell/PowerShell.git"
+
+# Install dependencies
+RUN yum install -y \
+        curl \
```
Comment (TravisEz13) [inline docker/tests/Templates/fxdependent-centos7/Dockerfile]: Don't add `curl`, see comments about `wget`

**Review by TravisEz13 — CHANGES_REQUESTED** [review]

File: docker/tests/Templates/fxdependent-centos7/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        glibc-locale-source \
+        git
+
+# Install dotnet-runtime
+RUN rpm -Uvh https://packages.microsoft.com/config/rhel/7/packages-microsoft-prod.rpm
+RUN yum install -y \
```
Comment (adityapatwardhan) [inline docker/tests/Templates/fxdependent-centos7/Dockerfile]: This wont be needed if using the 2.1-sdk or  2.1-runtime image is used.

File: docker/tests/Templates/fxdependent-debian.8/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+# Install dotnet-runtime
```
Comment (adityapatwardhan) [inline docker/tests/Templates/fxdependent-debian.8/Dockerfile]: This might not be needed.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: docker/tests/Templates/fxdependent-centos7/Dockerfile (hunk 1 above)
Comment (anmenaga) [reply docker/tests/Templates/fxdependent-centos7/Dockerfile]: fixed.

File: docker/tests/Templates/fxdependent-centos7/Dockerfile (hunk 2 above)
Comment (anmenaga) [reply docker/tests/Templates/fxdependent-centos7/Dockerfile]: Updated this for ubuntu18.04 and debian.9;
Unfortunately they don't have any other images.

File: docker/tests/Templates/fxdependent-debian.8/Dockerfile (hunk 3 above)
Comment (anmenaga) [reply docker/tests/Templates/fxdependent-debian.8/Dockerfile]: updated.

File: docker/tests/Templates/fxdependent-centos7/Dockerfile (hunk 4 above)
Comment (anmenaga) [reply docker/tests/Templates/fxdependent-centos7/Dockerfile]: updated.

File: docker/tests/Templates/fxdependent-centos7/Dockerfile (hunk 5 above)
Comment (anmenaga) [reply docker/tests/Templates/fxdependent-centos7/Dockerfile]: Updated base image for ubuntu18.04 and debian.9;
Unfortunately .NET doesn't have any other images.

File: docker/tests/Templates/fxdependent-debian.8/Dockerfile (hunk 6 above)
Comment (anmenaga) [reply docker/tests/Templates/fxdependent-debian.8/Dockerfile]: I didn't find .NET image for debian.8

**Comment by anmenaga** [comment]
@TravisEz13 @adityapatwardhan please take a look again.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@TravisEz13  Please have another look.

**Review by TravisEz13 — APPROVED** [review]

File: docker/tests/Templates/fxdependent-centos7/Dockerfile (hunk 2 above)
Comment (TravisEz13) [reply docker/tests/Templates/fxdependent-centos7/Dockerfile]: Do we need to test all these platforms?

**Comment by adityapatwardhan** [comment]
@TravisEz13 @anmenaga There are issues reported by CodeFactor for Docker files. Should they be resolved?

**Pull request merged**